staking-pool/src/lib.rs

```diff
@@ -70,111 +95,354 @@ impl StakingContract {
     /// Call to initialize the contract.
     /// Specify which account can change the staking key and the initial staking key with ED25519 curve.
     #[init]
-    pub fn new(owner: AccountId, stake_public_key: String) -> Self {
-        assert!(
-            env::state_read::<StakingContract>().is_none(),
-            "Already initialized"
-        );
-        let mut pk = vec![0];
-        pk.extend(bs58::decode(stake_public_key).into_vec().unwrap());
+    pub fn new(
+        owner_id: AccountId,
+        stake_public_key: Base58PublicKey,
+        max_number_of_seats: u32,
```

Comment:
agree with @jimmy3dita let's not call this seats.
I also not sure we really need this?
May be instead just have field with min stake? This min stake should cover the storage of the delegation?
Also ppl can't really make contract to overfill, because it will just decline if they send a transaction that takes more space then adds funds. So it kinda was working automatically having min stake requirement if there is too much space taken by contract?

I'm not completely sure how you are saving the space here with the active/inactive/archived here?
It seems like archived accounts still need to store accountid & account?

Reply:
The issue the iteration within a persistent map. Every lookup/insert requires `storage_read/storage_write`. These operations take too much gas comparing to the on time large `storage_read` because of the `base` fee required to locate the key.

This implementation doesn't issue constant amount of storage read/writes